ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/Footer.kt

```diff
@@ -154,3 +171,52 @@ private fun FooterPreview() {
         viewModel = MockViewModel(TestData.template2Offering)
     )
 }
+
+@Composable
+fun RowScope.AdaptiveView(views: List<@Composable () -> Unit>) {
+    var selectedIndex by remember { mutableStateOf(0) }
+    var maxSize by remember { mutableStateOf(0) }
+    var viewSizes by remember { mutableStateOf(List(views.size) { 0 }) }
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .onGloballyPositioned { coordinates ->
+                maxSize = coordinates.size.width
+            }
+            .align(Alignment.CenterVertically)
+            .background(Color.Green.copy(alpha = 0.5f))
+    ) {
+        // Offscreen measurement
```

Comment:
I was wondering, could we move calculating the items sizes to a different composable function? Maybe even calculating the index of the view we need to use.

Another concern is that this is adding a bunch of composables to the tree that are not really needed.. I was thinking of ways to avoid this except when measuring... I was thinking, we could call that composable function if we haven't selected an index... Though in that case, we would need to recalculate when the screen size changes (like changing orientation, or on foldable devices). So not sure if that might overcomplicate things.

Reply:
Looks like this performs pretty well 🤷🏻 Honestly I don't think it's a huge performance issue for just a few texts, but let me know what you think.